plugins/UM3NetworkPrinting/src/UM3OutputDevicePlugin.py

```diff
@@ -473,7 +473,16 @@ def _onCloudPrintingConfigured(self) -> None:
         # Set the machine's cloud flow as complete so we don't ask the user again and again for cloud connected printers
         active_machine = self._application.getMachineManager().activeMachine
         if active_machine:
-            active_machine.setMetaDataEntry("do_not_show_cloud_message", True)
+
+            # The active machine _might_ not be the machine that was in the added cloud cluster and
+            # then this will hide the cloud message for the wrong machine. So we only set it if the
+            # host names match between the active machine and the newly added cluster
+            saved_host_name = active_machine.getMetaDataEntry("um_network_key", "").split('.')[0]
+            added_host_name = device.toDict()["host_name"]
```

Comment:
Does this also have to match printers by IP address too?

Reply:
Shit, I guess it should... since manually added printers use the IP as the host name.

In this case I'm not really sure what to do. In theory every printer should just have a UUID tied to it, but that means a lot of infrastructural changes...